### ISSUE TITLE
Remove onHooks?

### ISSUE THREAD

**Issue by thejameskyle** [thejameskyle/backbone-routing#1]
Triggers are useful for external objects and special subclassing cases. However there's not a huge benefit of:

```js
Route.extend({
  fetch(id) {
    return this.store.find(id);
  },
  onFetch() {
    // do something...
  }
});
```

over:

```js
Route.extend({
  fetch(id) {
    return this.store.find(id).then(() => {
      // do something...
    });
  }
});
```

The later simplifies this codebase and shrinks the public api a lot. It also seems more obvious.

**Comment by ianmstew** [comment]
I’m not the router expert, but I’m all for a simpler API.  I also think `.then()` is a pattern everyone should know or learn, especially for handling the result of a specific data request.  Plus Backbone doesn't use "on" methods itself.

**Comment by jmeas** [comment]
In ember, there are nuanced differences between all of their hooks to help account for the numerous edge cases when routing. For instance, their `model` hook won't be called if you navigate to a route programmatically and pass a model in, while the `beforeModel` and `afterModel` hooks still will be called.

This router doesn't seem to have all of those edge cases covered – which is fine – but if there's no distinction between the different hooks, then they might as well be removed. In addition to reducing public surface API and underlying LoC, it will make the code that developers write easier to read, as it removes a layer of unnecessary abstraction.

tl;dr, sure, remove it.
